layers/+lang/python/funcs.el

```diff
@@ -68,3 +68,76 @@ when this mode is enabled since the minibuffer is cleared all the time."
 (defun spacemacs//python-imenu-create-index-use-semantic ()
   "Use semantic if the layer is enabled."
   (setq imenu-create-index-function 'semantic-create-imenu-index))
+
+(defun python-get-main-testrunner ()
+  "Get the main test runner."
+  (cond ((eq 'pytest python-test-runner) 'pytest)
+        ((and (listp python-test-runner)
+              (eq 'pytest (car python-test-runner))) 'pytest)
+        ((eq 'nose python-test-runner) 'nose)
+        ((and (listp python-test-runner)
+              (eq 'nose (car python-test-runner))) 'nose)))
```

Comment:
It would not be that safe since there is not default condition. I just copied it from the previous version of the PR without thinking much about it. It is updated to your simpler version now.